src/components/ConfidentialQueuePanel.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { ListGroupItem } from 'reactstrap'
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faSpinner } from '@fortawesome/free-solid-svg-icons'
+
+/* eslint-disable react/prefer-stateless-function */
+class ConfidentialQueuePanel extends React.Component {
+  render() {
+    let questionId
+    let position = 0
+    if (this.props.queue && this.props.queue.questions) {
+      this.props.queue.questions.forEach(qId => {
+        const question = this.props.questions[qId]
+        if (this.props.userId === question.askedById) {
+          questionId = qId
+        } else {
+          position += 1
+        }
+      })
+
+      const numQuestions = this.props.queue.questions.length
+      if (numQuestions === 0) {
+        return null
+      }
+      const cardTotalQuestionsMessage =
+        numQuestions === 1
+          ? 'There is 1 person on the queue'
+          : `There are ${numQuestions} people on the queue`
+      const singularOrPluralPerson = position === 1 ? 'person' : 'people'
+      const cardMessage = questionId
```

Comment:
This message isn't updating correctly for me. Steps to see the problem:

* Open a browser and log in as a student account
* Add a question: note that the message in that windows says "There is 1 person on the queue with 0 people in front of you." This is correct.
* Open a new browser and log in as another student
* Add a question: note that the message in that windows says "There are 2 people on the queue with 1 person in front of you." This is correct.
* Go back the the original window. The message says "There are 2 people on the queue with 1 person in front of you." This is incorrect, as that user was the first person on the queue and there should be no one in front of them.

Reply:
Found out I was doing a foreach over the questions and was never quitting when the user question was found... this should now be fixed